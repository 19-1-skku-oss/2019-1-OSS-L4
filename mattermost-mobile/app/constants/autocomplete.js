// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const AT_MENTION_REGEX = /\B(@([^@\r\n\s]*))$/i;

export const AT_MENTION_SEARCH_REGEX = /\bfrom:\s*(\S*)$/i;

export const CHANNEL_MENTION_REGEX = /\B(~([^~\r\n]*))$/i;

export const CHANNEL_MENTION_SEARCH_REGEX = /\b(?:in|channel):\s*(\S*)$/i;

export const DATE_MENTION_SEARCH_REGEX = /\b(?:on|before|after):\s*(\S*)$/i;

export const ALL_SEARCH_FLAGS_REGEX = /\b\w+:/g;