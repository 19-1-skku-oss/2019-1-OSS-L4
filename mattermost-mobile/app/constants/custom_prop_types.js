// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import PropTypes from 'prop-types';

export const Children = PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf([PropTypes.node])]);

export const Style = PropTypes.oneOfType([
    PropTypes.object, // inline style
    PropTypes.number, // style sheet entry
    PropTypes.array,
]);

export default {
    Children,
    Style,
};
