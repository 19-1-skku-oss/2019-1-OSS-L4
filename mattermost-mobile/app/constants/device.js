// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeviceInfo from 'react-native-device-info';
import RNFetchBlobFS from 'rn-fetch-blob/fs';
import keyMirror from 'mattermost-redux/utils/key_mirror';

const deviceTypes = keyMirror({
    CONNECTION_CHANGED: null,
    DEVICE_DIMENSIONS_CHANGED: null,
    DEVICE_TYPE_CHANGED: null,
    DEVICE_ORIENTATION_CHANGED: null,
    STATUSBAR_HEIGHT_CHANGED: null,
});

export default {
    ...deviceTypes,
    DOCUMENTS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Documents`,
    IMAGES_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Images`,
    IS_IPHONE_X: DeviceInfo.getModel().includes('iPhone X'),
    IS_TABLET: DeviceInfo.isTablet(),
    VIDEOS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Videos`,
};
