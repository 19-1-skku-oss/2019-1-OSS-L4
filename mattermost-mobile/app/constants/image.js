// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
