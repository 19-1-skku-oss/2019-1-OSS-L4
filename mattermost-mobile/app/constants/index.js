// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import PermissionTypes from './permissions';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    PermissionTypes,
    UpgradeTypes,
    ViewTypes,
};
