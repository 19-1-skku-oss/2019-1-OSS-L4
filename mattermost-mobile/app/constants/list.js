// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const VISIBILITY_CONFIG_DEFAULTS = {
    viewAreaCoveragePercentThreshold: 3,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
