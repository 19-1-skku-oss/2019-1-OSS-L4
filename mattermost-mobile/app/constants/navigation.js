// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
});

export default NavigationTypes;
