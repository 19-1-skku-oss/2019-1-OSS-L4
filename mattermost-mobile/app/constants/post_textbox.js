// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_CONTENT_HEIGHT = 100;
export const MAX_FILE_COUNT = 5;
export const IS_REACTION_REGEX = /(^\+:([^:\s]*):)$/i;
export const INSERT_TO_DRAFT = 'insert_to_draft';
export const INSERT_TO_COMMENT = 'insert_to_comment';
