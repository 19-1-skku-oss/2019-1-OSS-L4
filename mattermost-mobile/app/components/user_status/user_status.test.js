// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {General} from 'mattermost-redux/constants';
import Preferences from 'mattermost-redux/constants/preferences';

import UserStatus from './user_status';

describe('UserStatus', () => {
    const baseProps = {
        size: 32,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot, should default to offline status', () => {
        const wrapper = shallow(
            <UserStatus {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, away status', () => {
        const wrapper = shallow(
            <UserStatus
                {...baseProps}
                status={General.AWAY}
            />
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, dnd status', () => {
        const wrapper = shallow(
            <UserStatus
                {...baseProps}
                status={General.DND}
            />
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, online status', () => {
        const wrapper = shallow(
            <UserStatus
                {...baseProps}
                status={General.ONLINE}
            />
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
