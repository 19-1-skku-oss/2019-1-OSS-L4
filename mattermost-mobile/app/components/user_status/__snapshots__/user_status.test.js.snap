// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStatus should match snapshot, away status 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../dist/assets/images/status/away.png",
    }
  }
  style={
    Object {
      "height": 32,
      "tintColor": "#ffbc42",
      "width": 32,
    }
  }
/>
`;

exports[`UserStatus should match snapshot, dnd status 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../dist/assets/images/status/dnd.png",
    }
  }
  style={
    Object {
      "height": 32,
      "tintColor": "#f74343",
      "width": 32,
    }
  }
/>
`;

exports[`UserStatus should match snapshot, online status 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../dist/assets/images/status/online.png",
    }
  }
  style={
    Object {
      "height": 32,
      "tintColor": "#06d6a0",
      "width": 32,
    }
  }
/>
`;

exports[`UserStatus should match snapshot, should default to offline status 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../dist/assets/images/status/offline.png",
    }
  }
  style={
    Object {
      "height": 32,
      "tintColor": "rgba(61,60,64,0.3)",
      "width": 32,
    }
  }
/>
`;
